pinot-segment-local/src/main/java/org/apache/pinot/segment/local/io/writer/impl/MutableOffHeapByteArrayStore.java

```diff
@@ -178,7 +178,9 @@ public MutableOffHeapByteArrayStore(PinotDataBufferMemoryManager memoryManager,
       int numArrays, int avgArrayLen) {
     _memoryManager = memoryManager;
     _allocationContext = allocationContext;
-    _startSize = numArrays * (avgArrayLen + 4); // For each array, we store the array and its startoffset (4 bytes)
+    int estimatedSize =
+        numArrays * (avgArrayLen + 4); // For each array, we store the array and its startoffset (4 bytes)
```

Comment:
We need to make it more robust by using long to calculate the result and then cap it at `Integer.MAX_VALUE`. Currently if it is larger than 4G, we will end up with a very small start size.